sdk/python/agent_protocol/middlewares.py

```diff
@@ -11,3 +11,35 @@ async def not_found_exception_handler(
         str(exc),
         status_code=404,
     )
+
+
+class AgentMiddleware:
+    """
+    Middleware that injects the agent instance into the request scope.
+    """
+
+    def __init__(self, app: FastAPI, agent: "Agent"):
+        """
+
+        Args:
+            app: The FastAPI app - automatically injected by FastAPI.
+            agent: The agent instance to inject into the request scope.
+
+        Examples:
+            >>> from fastapi import FastAPI, Request
+            >>> from agent_protocol.agent import Agent
+            >>> from agent_protocol.middlewares import AgentMiddleware
+            >>> app = FastAPI()
+            >>> @app.get("/")
+            >>> async def root(request: Request):
+            >>>     agent = request["agent"]
+            >>>     return agent
```

Comment:
Isn't this confusing? Why are you returning agent?

Reply:
The returning the agent in the example is confusing will update